### PR TITLE
feat(timescaledb): create a TimescaleDB documentation

### LINKED ISSUE #1511: [PostgreSQL] Document the TimescaleDB extension

### LINKED ISSUE #1512: [TimescaleDB] Document how to downsample and limit the data retention
https://docs.timescale.com/timescaledb/latest/how-to-guides/user-defined-actions/example-downsample-and-compress/#downsample-and-compress

### PR DESCRIPTION
**Pull request by curzolapierre**
Contains:
- TimescaleDB extension introduction
- Downsampling
- Data retention

Fix #1511 
Fix #1512

### CODE REVIEW

File: _posts/databases/postgresql/TimescaleDB/2000-01-01-downsample.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+# Example
+
+Bellow an example of a logic of downsampling. It is based on TimescaleDB documentation.
+
+The datas are stored in a table `conditions`, composed by `temperature` and `humidity`
+value. This is a use case where a table is composed of raw datas.
+
+Now we want to decrease the database size. We don't need to have data by the
+minute for data older than one week, the average of values by hour is enough.
+So we can use a downsample logic which will reduce the datas older than one week,
```
Comment (Soulou) [inline _posts/databases/postgresql/TimescaleDB/2000-01-01-downsample.md:22]:
```suggestion
So we can use a downsampling logic which will reduce the datas older than one week,
```

**Review by Soulou — CHANGES_REQUESTED** [review]

**Comment by Soulou** [comment]
`downsample` is a verb, not a noun ;-)

File: _posts/databases/postgresql/TimescaleDB/2000-01-01-downsample.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+}
+```
+
+The scheduled task will be executed every day at midnight. And execute a `psql`
+command that invoke a SQL script called `call_procedure.sql`.
+
+Create this SQL script file at the root of your application's source code containing
+the code below:
+
+```SQL
```
Comment (EtienneM) [inline _posts/databases/postgresql/TimescaleDB/2000-01-01-downsample.md:120]: I don't know if it matters for Jekyll but I would write a lower case `SQL`.

File: _posts/databases/postgresql/TimescaleDB/2000-01-01-downsample.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+call downsample_conditions(config => '{"lag":"3 week", "period":"1 hour"}');
+```
+
+This script invoke the created procedure `downsample_conditions` to apply it for
```
Comment (EtienneM) [inline _posts/databases/postgresql/TimescaleDB/2000-01-01-downsample.md:124]:
```suggestion
This script invokes the created procedure `downsample_conditions` to apply it for
```

File: _posts/databases/postgresql/TimescaleDB/2000-01-01-intro.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+title: TimescaleDB PostgreSQL Extensions
+nav: Extensions
+modified_at: 2022-01-21 00:00:00
+tags: timescale databases postgresql extensions
+index: 1
+---
+
+TimescaleDB is a PostgreSQL extension for the support of the time-series data.
+The version available on Scalingo is the Open-source edition, which is different
+than the Community edition. You can check the comparison between both edition,
```
Comment (EtienneM) [inline _posts/databases/postgresql/TimescaleDB/2000-01-01-intro.md:11]: By being an external reader, I don't understand what you are talking about with this "Community edition". It seems it comes out of nowhere. Maybe it's less odd to say that the list of TimescaleDB functions introduced by enabling TimescaleDB on a PostgreSQL database is here (https://docs.timescale.com/api/latest/). From this list, the functions labelled as `Community` are not available on Scalingo.

**Review by EtienneM — CHANGES_REQUESTED** [review]

File: _posts/databases/postgresql/timescaledb/2000-01-01-intro.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+title: TimescaleDB
+nav: Introduction
+modified_at: 2022-01-21 00:00:00
+tags: timescale databases postgresql extensions
+index: 1
+---
+
+TimescaleDB is a PostgreSQL extension for the support of the time-series data.
+This extension add several features and functions, you can find more information
+on their documentation: [https://docs.timescale.com/api/latest](https://docs.timescale.com/api/latest).
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-intro.md:11]: I find it ugly to have the URL displayed, I prefer a link with a meaningful text
```suggestion
TimescaleDB is a PostgreSQL extension for the support of time-series data.
This extension adds several features and functions. You can find more information
on the [official documentation](https://docs.timescale.com/api/latest).
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-intro.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+There are multiple licences of TimescaleDB, __Community edition__ and
+__Open-source edition__. The latter is the only one available on Scalingo, hence
+TimescaleDB's functions labeled "community" on their documentation
+are not available on the platform.
+
+Several important features for time-series databases not available on the
+Open-source edition. It is still possible without the functions to put them in
+place, we present some of them here.
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-intro.md:20]: I don't like this paragraph. I find it too deceptive. I don't see what it adds, I would remove it
```suggestion
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-intro.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+---
+
+TimescaleDB is a PostgreSQL extension for the support of the time-series data.
+This extension add several features and functions, you can find more information
+on their documentation: [https://docs.timescale.com/api/latest](https://docs.timescale.com/api/latest).
+
+There are multiple licences of TimescaleDB, __Community edition__ and
+__Open-source edition__. The latter is the only one available on Scalingo, hence
+TimescaleDB's functions labeled "community" on their documentation
+are not available on the platform.
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-intro.md:16]: Maybe this introduction page should talk about the two other pages created, and link to them?

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
+---
+title: Downsampling
+nav: Downsampling
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:3]: You don't need a nav if it's the same as the title:
```suggestion
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+modified_at: 2022-01-21 00:00:00
+tags: timescale databases postgresql extensions downsampling
+---
+
+Downsampling data allows to reduce the amount of data by simplifying the old values.
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:8]: Simplification suggestion:

```suggestion
The downsampling reduces the amount of data by simplifying the old values.
```

I don't like the word "simplifying". I don't understand what it means

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+
+## Example
+
+Bellow an example of a logic of downsampling. It is based on TimescaleDB documentation.
+
+The data is stored in a table `conditions`, composed by `temperature` and `humidity`
+value. This is a use case where a table is composed of raw datas.
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:15]: data is still uncountable. Do you have a dictionary installed in the text editor you use to write doc? It should highlight that `datas` does not exist

```suggestion
value. This is a use case where a table is composed of raw data.
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+Now we want to decrease the database size. We don't need to have data by the
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:17]: nitpick: I'm not sure about the purpose of the word `Now` here. From my point of view it does not add any information.

```suggestion
We want to decrease the database size. We don't need to have data by the
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+minute for data older than one week, the average of values by hour is enough.
+So we can use a downsampling logic which will reduce the datas older than one week,
+and only store the average by hour.
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:20]: We rarely use the future tense in documentation pages:
```suggestion
So we can use a downsampling logic which reduces the data older than one week,
and only stores the average by hour.
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+
+
+### Datas
+
+For the following table:
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:25]: Not sure about this one

```suggestion
For the following data:
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+-------------------------------+----------+--------------------+--------------------
+ 2022-01-21 09:50:05.049911+00 | office   |                 75 |                 52
+ 2022-01-21 09:50:05.058412+00 | office   |                 70 |                 57
+ 2022-01-21 09:50:05.059489+00 | office   |                 76 |                 52
+ 2022-01-20 09:50:05.06032+00  | office   |                 79 |                 54
+```
+
+### Downsample Procedure
+
+The procedure bellow contains the logic to downsample the table `conditions`
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:39]: I didn't know but the word `bellow` exists and means "mugir" or "beugler" xD

```suggestion
The procedure below contains the logic to downsample the table `conditions`
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+nav: Downsampling
+modified_at: 2022-01-21 00:00:00
+tags: timescale databases postgresql extensions downsampling
+---
+
+Downsampling data allows to reduce the amount of data by simplifying the old values.
+
+## Example
+
+Bellow an example of a logic of downsampling. It is based on TimescaleDB documentation.
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:12]: If it's based on the official doc, please make `TimescaleDB documentation` a link to the said page

```suggestion
Below an example of a logic of downsampling. It is based on TimescaleDB documentation.
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+  END LOOP;
+END
+$$;
+```
+
+### Execute Procedure
+
+You can schedule the execution of the previous procedure by using
+[Scalingo scheduler]({% post_url platform/app/task-scheduling/2000-01-01-scalingo-scheduler %}).
+To do so, add a `cron.json` file at the root of your application’s source code,
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:97]: nitpick: not sure `To do so` adds anything here. And it really sounds like a French translation to my hears. I might be wrong:
```suggestion
Add a `cron.json` file at the root of your application’s source code,
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md [hunk 17]
```diff
@@ -0,0 +1,66 @@
+---
+title: Data Retention
+nav: Data Retention
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md:3]: Duplicate of the title, not needed 
```suggestion
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+modified_at: 2022-01-21 00:00:00
+tags: timescale databases postgresql extensions data retention
+---
+
+A data retention policy aims at removing data older than a certain duration from the database.
+
+# Example
+
+Below an example of a retention policy which apply on __all__ hypertables.
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md:12]:
```suggestion
Below an example of a retention policy which applies on __all__ hypertables.
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+It could possible to specify tables by changing the `format` or add filter using
+`WHERE` clause in the `PERFORM` query.
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md:14]: I don't understand this sentence

File: _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+  PERFORM drop_chunks(format('%I.%I', table_schema, table_name), older_than => drop_after)
+    FROM timescaledb_information.hypertables;
+END
+$$;
+```
+
+
+## Execute Procedure
+
+You can schedule the execution of the previous procedure by using
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md:39]:
```suggestion
You can schedule the execution of the previous procedure by using the
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md [hunk 21]
```diff
@@ -0,0 +1,122 @@
...
+
+    COMMIT;
+  END LOOP;
+END
+$$;
+```
+
+### Execute Procedure
+
+You can schedule the execution of the previous procedure by using
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-downsampling.md:95]:
```suggestion
You can schedule the execution of the previous procedure by using the
```

File: _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+    {
+      "command": "0 0 * * * dbclient-fetcher psql && psql $SCALINGO_POSTGRESQL_URL -f data_retention_procedure.sql",
+      "size": "S"
+    }
+  ]
+}
+```
+
+The scheduled task will be executed every day at midnight. And execute a `psql`
+command that invoke a SQL script called `data_retention_procedure.sql`.
```
Comment (EtienneM) [inline _posts/databases/postgresql/timescaledb/2000-01-01-data-retention.md:56]:
```suggestion
command that invokes a SQL script called `data_retention_procedure.sql`.
```

**Review by EtienneM — CHANGES_REQUESTED** [review]
Thanks a lot for these pages. I didn't expect it to be that simple. I'm happy \o/

**Review by EtienneM — APPROVED** [review]

**Comment by EtienneM** [comment]
@curzolapierre we don't want auto merge on this PR. We don't want our documentation to talk about TimescaleDB whereas the product is not ready yet. After Léo approves this PR, we should wait for the end of the development of this product to merge the PR

**Review by Soulou — APPROVED** [review]
LGTM to merge after product release.

**Review by EtienneM — APPROVED** [review]

**Pull request merged**